lib/resources/resourceBase.js

```diff
@@ -64,7 +64,10 @@ var ResourceBase = function (endpoint, config) {
         }
 
         if (body && body.error) {
-          return reject(body.error);
+          var error = new Error(body.error.message + ' (' + body.error.status_code + ')');
+          error.statusCode = body.error.status_code;
+          error.statusMessage = body.error.message;
```

Comment:
This goes against what I told you on Friday, but I thought about it a bit more and it's probably better to use the `message` property of the Error object and then just add the `status_code` property. So this becomes:

```js
var error = new Error(body.error.message);
error.status_code = body.error.status_code;
return reject(error);
```

And then do the same for the 500 case:

```js
var error = new Error(resp.statusMessage);
error.status_code = resp.statusCode;
return reject(error);
```

While this changes how you'd handle a 500 (accessing `err.status_code` instead of `err.statusCode`), it:

* Doesn't change how you'd handle normal errors (both `err.message` and `err.status_code` still exist)
* Makes normal errors and 500s consistent

What do you think?